plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -104,12 +110,24 @@ public static void addMavenLibraries(IProject project, List<Library> libraries,
       }
       subMonitor.worked(1);
     }
-    javaProject.setRawClasspath(newRawClasspath.toArray(new IClasspathEntry[0]), subMonitor);
+    
+    if (addToClasspath) {
+      javaProject.setRawClasspath(newRawClasspath.toArray(new IClasspathEntry[0]), subMonitor);
+    }
     
     runContainerResolverJob(javaProject);
```

Comment:
Maybe we should add a method to `LibraryClasspathContainerResolverService` that takes a list of IClasspathEntry rather than a JavaProject?